toxcore/friend_requests.c

```diff
@@ -158,10 +158,10 @@ static int friendreq_handlepacket(void *object, const uint8_t *source_pubkey, co
 
     addto_receivedlist(fr, source_pubkey);
 
-    const uint32_t message_len = length - sizeof(fr->nospam);
-    VLA(uint8_t, message, message_len + 1);
+    const size_t message_len = (size_t)length - sizeof(fr->nospam);
+    uint8_t message[ONION_CLIENT_MAX_DATA_SIZE + 1];
     memcpy(message, packet + sizeof(fr->nospam), message_len);
-    message[SIZEOF_VLA(message) - 1] = 0; /* Be sure the message is null terminated. */
+    message[message_len] = 0; /* Be sure the message is null terminated. */
```

Comment:
This is a VLA. Even `constexpr` wouldn't save you here, because `length` is not a constexpr.